sdk/storage/Azure.Storage.DataMovement/src/CopyStatusHandler.cs

```diff
@@ -49,37 +49,44 @@ public CopyStatusHandler(Behaviors behaviors)
             _updateTransferStatus = behaviors.UpdateTransferStatus
                 ?? throw Errors.ArgumentNull(nameof(behaviors.UpdateTransferStatus));
 
-            AddStatusEvent();
+            _getStatusHandler += GetStatusEvent;
         }
 
-        public void AddStatusEvent()
+        public void Dispose()
         {
-            _getStatusHandler += async (CopyStatusEventArgs args) =>
+            Cleanup();
+        }
+
+        public void Cleanup()
+        {
+            _getStatusHandler -= GetStatusEvent;
+        }
+
+        public async Task GetStatusEvent(CopyStatusEventArgs args)
```

Comment:
nit: I see you turned the function `private` above. Did you want to do that for all of these?